app_flutter/lib/status_grid.dart

```diff
@@ -99,64 +99,162 @@ class StatusGrid extends StatelessWidget {
-            gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
-                crossAxisCount: columnCount),
-            itemBuilder: (BuildContext context, int gridIndex) {
-              if (gridIndex == 0) {
-                /// The top left corner of the grid is nothing since
-                /// the left column is for [CommitBox] and the top
-                /// row is for [TaskIcon].
-                return const SizedBox();
-              }
-
-              /// This [GridView] is composed of a row of [TaskIcon] and a subgrid
-              /// of [List<List<Task>>]. This allows the row of [TaskIcon] to align
-              /// with the column of [Task] that it maps to.
-              ///
-              /// Mapping [gridIndex] to [index] allows us to ignore the overhead the
-              /// row of [TaskIcon] introduces.
-              final int index = gridIndex - columnCount;
-              if (index < 0) {
-                return TaskIcon(task: taskMatrix.sampleTask(gridIndex - 1));
-              }
-
-              final int row = index ~/ columnCount;
-              if (index % columnCount == 0) {
-                return CommitBox(commit: statuses[row].commit);
-              }
-
-              final int column = (index % columnCount) - 1;
-              final Task task = taskMatrix.task(row, column);
-              if (task == null) {
-                /// [Task] was skipped so don't show anything.
-                return const SizedBox();
-              }
-
-              return TaskBox(
-                task: task,
-                buildState: buildState,
-              );
-            },
+      child: ListView(
+        children: rows,
+        shrinkWrap: false,
+      ),
+    );
+  }
+
+  List<Widget> _buildRows() {
+    final List<Widget> rows = <Widget>[];
+
+    rows.add(
+      Container(
+        height: 50,
+        child: NotificationListener<ScrollNotification>(
+          child: ListView(
+            controller: controllers[0],
+            children: _buildTaskIconRow(),
+            scrollDirection: Axis.horizontal,
           ),
+          onNotification: (ScrollNotification scrollInfo) {
+            _syncScroller.processNotification(scrollInfo, controllers[0]);
+            return;
+          },
         ),
       ),
     );
+
+    for (int rowIndex = 0; rowIndex < widget.taskMatrix.rows; rowIndex++) {
+      final List<TaskBox> tasks = <TaskBox>[];
+      for (int colIndex = 0; colIndex < widget.taskMatrix.columns; colIndex++) {
+        tasks.add(TaskBox(
+          buildState: widget.buildState,
+          task: widget.taskMatrix.task(rowIndex, colIndex),
+        ));
+      }
+
+      rows.add(
+        Container(
+          height: 50,
+          child: NotificationListener<ScrollNotification>(
+            child: ListView.builder(
+              controller: controllers[rowIndex + 1],
+              scrollDirection: Axis.horizontal,
+              itemCount: widget.taskMatrix.columns + 1,
```

Comment:
Do  you know the itemExtent for both controllers too?

Reply:
I didn't know about this, so thanks for sharing! Added.